wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -692,6 +777,34 @@ instance BuildableSafeGen Account where
 instance Buildable [Account] where
     build = bprint listJson
 
+-- | An external wallet 'ExternalAccount'.
+data ExternalAccount = ExternalAccount
+  { accExtIndex     :: !AccountIndex
+  , accExtAddresses :: [V1 Core.Address]  -- should be WalletAddress
+  , accExtAmount    :: !(V1 Core.Coin)
+  , accExtName      :: !Text
+  , accExtWalletId  :: WalletId
```

Comment:
My idea was simple: if we have separate types `Wallet` and `ExternalWallet`, so _probably_ we need `ExternalAccount/ExternalAddress` as well. No?

Reply:
My gut feeling is that is always better to start with _less_ types and create new ones only if you are absolutely cornered and cannot accomplish what we want with the types you have. It taps into the simplicity you also advocated in one of your talks 😉 The less types, the less code, the less conditional branching, the better!

For a `Wallet` the distinction with an `External` is somehow obvious and inevitable because we need to construct them differently and store different things in them, but for the rest of the types in the HD hierarchy, this might not be true, and it would be nice if we manage to retain the old types 😀